setup.py

```diff
@@ -93,34 +93,45 @@
     include_package_data=True,
     install_requires=[
         "backports.cached_property ; python_version<'3.8'",
-        "click>=8.1.0",
-        "ethpm-types>=0.3.2,<0.4.0",
-        "eip712>=0.1.0,<0.2",
-        "evm-trace>=0.1.0.a6",
+        "click>=8.1.0,<9",
+        "ijson>=3.1.4,<4",
+        "IPython>=7.31.1,<8",
         "hexbytes>=0.2.2,<1.0.0",
+        "importlib-metadata",
         "packaging>=20.9,<21.0",
         "pandas>=1.3.0,<2.0",
         "pluggy>=0.13.1,<1.0",
         "pydantic>=1.9.0,<2.0",
         "PyGithub>=1.54,<2.0",
         "pygit2>=1.7.2,<2.0",
         "pyyaml>=0.2.5",
-        "py-geth>=3.6.0",
         "requests>=2.28.1,<3.0",
-        "importlib-metadata",
-        "singledispatchmethod ; python_version<'3.8'",
-        "IPython>=7.31.1",
-        "pytest>=6.0,<8.0",
+        "packaging>=20.9,<21",
+        "pandas>=1.3.0,<2",
+        "pluggy>=0.13.1,<1",
+        "pydantic>=1.9.0,<2",
+        "PyGithub>=1.54,<2",
+        "pygit2>=1.7.2,<2",
+        "pytest>=6.0,<8",
+        "python-dateutil>=2.8.2,<3",
+        "pyyaml>=6.0,<7",
+        "requests>=2.25.1,<3",
```

Comment:
How many of these are required dependencies?

Reply:
AFAIK, all of them in this `install_requires` list are required dependencies.
Note: I did not add any dependencies in this PR but I reorganized them such that the Ethereum foundation and Apeworx dependencies are grouped. The remaining dependencies are alphabetized so it is easier to lookup them up (as the list is quite long).

Reply:
Maybe... create a ticket to review dependencies lol

Reply:
I will do it now instead:

Packages removes:

* duplicate `hexbytes` entry
* `importlib-metadata` - not sure if this is needed, but it is part of the template. Will put back if you say it's needed or the build fails.
* `eth-rlp`: didn't see it being used anywhere so removed it

Packages kept and reason for their usage:

* `backports.cached_property`: used wherever `@cached_property` is used
* `click`: The CLI framework we use
* `ijson`: Used by `ape-geth` for streaming logs
* `IPython`: The lib that powers `ape console`
* `importlib-metadata`: Required until we no longer support 3.7
* `packaging`: Used for parsing versions in the package manifest
* `pandas:` Used for querying
* `pluggy`: Used for plugin system
* `pydantic`: Used for API classes
* `PyGithub`: Used to talk to Github for downloading dependencies or installing plugins
* `pygit2`: Used as a `git` wrapper for installing dependencies
* `pyyaml`: Used to parse config file
* `requests`: Used in `ape-geth` for request streaming as well as checking off exceptions
* `rich`: Used for displaying trees in the terminal
* `singledispatchmethod`: Used for overloading methods with varying arg types
* `tqdm`: Used for CLI progress bars
* `typing-extensions`: Used to assist installing the tool correctly, can remove after python 3.7.
* `eth-abi`: Used for ABI parsing
* `eth-account`: Used for accounts in ape, like verifying signatures
* `eth-utils`: tools for working with Ethereum, used everywhere
* `hexbytes`: class for working with bytes data types in Ethereum
* `py-geth`: used for the ephemeral local geth process
* `web3[tester]`: Used as Ethereum client
* `eip712`: Uses a type when checking the signature for compatibility purposes with the library
* `ethpm-types`: Used for contract types and package manifest stuff
* `evm-trace`: Used for transaction tracing

Reply:
perfect! thank you!